### ISSUE TITLE
'not null' not reported for anonymous access discriminant

### ISSUE THREAD

**Issue by simonjwright** [simonjwright/asis2xml#2]
Given
```
package Accessing is
   type T (Cons: not null access Integer) is null record;
end Accessing;
```
we get
```
<known_discriminant_part>
  <discriminant_specification>
    <defining_identifier>Cons</defining_identifier>
    <access_definition kind="anonymous_access_to_variable">
      <identifier>Integer</identifier>
    </access_definition>
  </discriminant_specification>
</known_discriminant_part>
```
whereas for a named access type `P` we get
```
<known_discriminant_part>
  <discriminant_specification not_null="true">
    <defining_identifier>Cons</defining_identifier>
    <identifier>P</identifier>
  </discriminant_specification>
</known_discriminant_part>
```

**Comment by simonjwright** [comment]
This also applies to variable declarations. There’s considerable scope for confusion, not least because `not null` occurs in multiple places in the syntax, so that sometimes it’s a bit arbitrary which is chosen - or even both, as in
```
procedure F (X : not null access Integer);
```
which gives
```
<procedure_declaration>
  <defining_identifier>F</defining_identifier>
  <parameter_specification not_null="true">
    <defining_identifier>X</defining_identifier>
    <access_definition kind="anonymous_access_to_variable"
    not_null="true">
      <identifier>Integer</identifier>
    </access_definition>
  </parameter_specification>
</procedure_declaration>
```

**Comment by simonjwright** [comment]
Fixed in c54724f1274813dfee1f7ca9c8c20cc352ac3bff
